snapshots/snapshot_reporter/src/snapshot_reporter.py

```diff
@@ -0,0 +1,132 @@
+"""
+What do?
+
+"""
+
+import boto3
+from elasticsearch import Elasticsearch
+import httpx
+import humanize
+
+from dateutil import parser
+import json
+import os
+
+
+def get_secret(secret_id):
+    """
+    Use secrets from SecretsManager to construct an Elasticsearch client.
+    """
+    secrets_client = boto3.client("secretsmanager")
+
+    resp = secrets_client.get_secret_value(SecretId=secret_id)
+
+    try:
+        # The secret response may be a JSON string of the form
+        # {"username": "…", "password": "…", "endpoint": "…"}
+        secret = json.loads(resp["SecretString"])
+    except ValueError:
+        secret = resp["SecretString"]
+
+    return secret
+
+
+def get_elastic_client(elastic_secret_id):
+    secret = get_secret(elastic_secret_id)
+
+    return Elasticsearch(
+        secret["endpoint"], http_auth=(secret["username"], secret["password"])
+    )
+
+
+def get_snapshots(es_client, days_to_fetch):
+    index = "snapshots"
+
+    date_filter = [
+        {"range": {"snapshotJob.requestedAt": {"gte": f"now-{days_to_fetch}d/d"}}}
+    ]
+    es_query = {"bool": {"filter": date_filter}}
+    sort_by = [{"snapshotJob.requestedAt": {"order": "desc"}}]
+
+    response = es_client.search(index=index, body={"query": es_query, "sort": sort_by})
+
+    return [hit["_source"] for hit in response["hits"]["hits"]]
+
+
+def prepare_slack_payload(snapshots, days_to_fetch):
+    def _snapshot_message(snapshot):
+        index_name = snapshot["snapshotResult"]["indexName"]
+        document_count = snapshot["snapshotResult"]["documentCount"]
+        s3_size = snapshot["snapshotResult"]["s3Size"]["bytes"]
+
+        requested_at = parser.parse(snapshot["snapshotJob"]["requestedAt"]).strftime(
+            "%A, %B %e, %I:%M %p"
+        )
+
+        started_at = parser.parse(snapshot["snapshotResult"]["startedAt"])
+
+        finished_at = parser.parse(snapshot["snapshotResult"]["finishedAt"])
+
+        return "\n".join(
+            [
+                f'*"{index_name}" on {requested_at}*',
+                f"{humanize.naturalsize(s3_size)} containing {humanize.intword(document_count)} documents ({document_count} exactly)",
```

Comment:
nit: consider adding commas for readability.

```suggestion
                f"{humanize.naturalsize(s3_size)} containing {humanize.intword(document_count)} documents ({humanize.intcomma(document_count)} exactly)",
```